data/frameReceiver/include/MercuryFrameDecoder.h

```diff
@@ -85,7 +85,8 @@ namespace FrameReceiver
 
       void* get_packet_header_buffer(void);
 
-      uint32_t get_frame_counter(void) const;
+      uint32_t get_32b_frame_counter(void) const;

```

Comment:
This 32b/64b terminology is confusing as it doesn't obviously relate to which packet header is in use. You also discard the difference by immediately casting to `int` as the variable `frame`. In fact `frame number` in the frame header is `uint32_t`, which means you will discard precision in the frame number in the shared memory buffer, and overflow/wrap the value if the frame number exceeds 2^32. I suggest simplifying this to a single `get_frame_counter` method which returns a `uint64_t` (upcasting in the case of the legacy header with a 32b frame counter). You can then modify the type of `frame_number` in the frame header to match, and also the type of any variables used in the decoder, e.g. `current_frame_seen_`.

The frame number that is eventually populated in the FP metadata associated with the frame is actually 64b, so this will improve end-to-end consistency.